packages/paste-website/src/components/shortcodes/component-header/index.tsx

```diff
@@ -2,160 +2,123 @@ import * as React from 'react';
-}> = ({name, ogImagePath, categoryRoute}) => (
-  <>
-    {ogImagePath ? (
-      <Helmet>
-        <meta property="og:image" content={getOpengraphServiceUrl(ogImagePath)} />
-      </Helmet>
-    ) : null}
-    <Breadcrumb>
-      <BreadcrumbItem to="/">Home</BreadcrumbItem>
-      <BreadcrumbItem to={categoryRoute}>{getCategoryNameFromRoute(categoryRoute)}</BreadcrumbItem>
-    </Breadcrumb>
-    <Heading as="h1" variant="heading10">
-      {getHumanizedNameFromPackageName(name)}
-    </Heading>
-  </>
+const IconAnchor: React.FC<{href: string; icon: React.ReactNode; children: string}> = ({href, icon, children}) => (
+  <Anchor href={href}>
+    <Box as="span">
+      <Box as="span" display="inline-block" marginRight="space20">
+        {icon}
+      </Box>
+      {children}
+    </Box>
+  </Anchor>
 );
 
-interface ComponentHeaderProps {
-  children?: React.ReactElement;
+export interface ComponentHeaderProps {
   name: string;
   categoryRoute: typeof SidebarCategoryRoutes[keyof typeof SidebarCategoryRoutes];
-  githubUrl: string;
-  storybookUrl: string;
-  data?: [
-    {
-      node: {
-        name: string;
-        category: string;
-        description: string;
-        status: string;
-        sideEffects: boolean;
-        version: string;
-      };
-    }
-  ];
-  packageStatus: PackageStatusObject;
-  isCore?: boolean;
+  description?: string;
+  designCommitteeReview?: string;
+  engineerCommitteeReview?: string;
+  figmaStatus?: string;
+  githubUrl?: string;
+  packageName?: string;
+  status?: string;
+  storybookUrl?: string;
+  version?: string;
 }
 
-export const PackageValue: React.FC = ({children}) => {
-  return (
-    <Text as="dd" display="inline-block">
-      {children}
-    </Text>
-  );
-};
-
-export const PackageLabel: React.FC = ({children}) => {
-  return (
-    <Text as="dt" display="inline-block" color="colorTextWeak" css={{width: '80px'}}>
-      {children}
-    </Text>
-  );
-};
-
-const PackageInstallSnippet: React.FC = ({children}) => {
-  return <InlineCode>{children}</InlineCode>;
-};
-
 const ComponentHeader: React.FC<ComponentHeaderProps> = ({
   name,
   categoryRoute,
+  description,
+  designCommitteeReview,
+  engineerCommitteeReview,
+  figmaStatus,
   githubUrl,
+  packageName,
+  status,
   storybookUrl,
-  data,
-  packageStatus,
-  isCore = false,
+  version,
 }) => {
-  if (data == null || data[0] == null || data[0].node == null) {
-    return <ComponentHeaderBasic categoryRoute={categoryRoute} name={name} />;
-  }
-  if (packageStatus == null || packageStatus[0] == null || packageStatus[0].node == null) {
-    return <ComponentHeaderBasic categoryRoute={categoryRoute} name={name} />;
-  }
+  const ogImagePath = packageName
+    ? `${categoryRoute.replace('/', '')}/${getNameFromPackageName(packageName)}`
+    : undefined;
+
+  const showStatus = status || figmaStatus || designCommitteeReview || engineerCommitteeReview;
```

Comment:
Personal preference: Would prefer to see this business logic in the component itself. The Status component API includes each of these as a prop on its interface. Since they are known, I think it makes sense to move this conditional into that component. That makes the return body of this component a lot easier to read and more clear.

Reply:
Updated.